File: src/dataset/typescript/acronym/acronym.ts
export const parse = (phrase: string): string => {
  let acronym = "";
  const matches = phrase.match(/[A-Z]+[a-z]*|[a-z]+/g) || [];

  matches.forEach((match) => acronym += match.charAt(0).toUpperCase())

  return acronym;
};

File: src/dataset/Java/acronym_Exercism_java_rabestro/Acronym.java
import java.util.function.Supplier;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

record Acronym(String phrase) implements Supplier<String> {
    private static final Pattern FIRST_LETTER_PATTERN =
            Pattern.compile("(?<![\\p{Alpha}'])\\p{Alpha}");

    public String get() {
        if (phrase == null || phrase.isEmpty()) {
            return "";
        }
        
        Matcher matcher = FIRST_LETTER_PATTERN.matcher(phrase);
        StringBuilder result = new StringBuilder();
        
        while (matcher.find()) {
            char ch = matcher.group().charAt(0);
            result.append(Character.toUpperCase(ch));
        }
        
        return result.toString();
    }
}